### DISCUSSION TITLE
How do I make a treeview node initially selected ?

### DISCUSSION THREAD

**Discussion by jiiteel** [enisn/UraniumUI#616] [Q&A]
Hi,

I tried to make a treeview node initially selected by binding the checkbox IsChecked-property to a property in the node. Tested with the UraniumUI sample solution but ended up with empty space inside the tree for the selected node and the actual node is hidden.

modified the treeview.xaml with
   <material:CheckBox Text="{Binding Name}" Margin="10" IsChecked="{Binding IsSelected}" >

and in TreeViewPageViewModel.cs modified one sample item
    new MyItem("A.1") { IsSelected = true },

and added observable property to the MyItem-class
    public virtual bool IsSelected { get => isSelected; set => SetProperty(ref isSelected, value); }

ended up with this. As you can seen the item A.1 is hidden (the space for it is still there)

<img width="135" alt="image" src="https://github.com/enisn/UraniumUI/assets/111491595/522a0c99-4c4e-41f2-be94-a7e2b80f211f">

**Comment by claybrooks** [comment]
> Hi,
> 
> 
> 
> I tried to make a treeview node initially selected by binding the checkbox IsChecked-property to a property in the node. Tested with the UraniumUI sample solution but ended up with empty space inside the tree for the selected node and the actual node is hidden.
> 
> 
> 
> modified the treeview.xaml with
> 
>    <material:CheckBox Text="{Binding Name}" Margin="10" IsChecked="{Binding IsSelected}" >
> 
> 
> 
> and in TreeViewPageViewModel.cs modified one sample item
> 
>     new MyItem("A.1") { IsSelected = true },
> 
> 
> 
> and added observable property to the MyItem-class
> 
>     public virtual bool IsSelected { get => isSelected; set => SetProperty(ref isSelected, value); }
> 
> 
> 
> ended up with this. As you can seen the item A.1 is hidden (the space for it is still there)
> 
> 
> 
> <img width="135" alt="image" src="https://github.com/enisn/UraniumUI/assets/111491595/522a0c99-4c4e-41f2-be94-a7e2b80f211f">
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

Having the same issue, unable to bind to the checkbox without seeing what you are seeing. Also getting null ref exceptions when clicking on anything within the checkbox.

Can we get some direction on how to set the initial check state for the treeview?

**Reply by claybrooks** [reply]
Opened issue: https://github.com/enisn/UraniumUI/issues/905
